src/meili/match_route.cc

```diff
@@ -208,69 +216,145 @@ bool MergeRoute(std::vector<EdgeSegment>& route, const State& source, const Stat
     throw std::logic_error("The first edge must be an origin (invalid predecessor)");
   }
 
-  MergeEdgeSegments(route, segments.rbegin(), segments.rend());
+  // MergeEdgeSegments(route, segments.rbegin(), segments.rend());
+  route.insert(route.end(), segments.crbegin(), segments.crend());
```

Comment:
so we only merge segments that were cut by a trace point landing on them IF that trace point was a break type. this allows multileg map matches